Code/Framework/AzFramework/Tests/QualitySystemComponentTests.cpp

```diff
@@ -75,11 +77,27 @@ namespace UnitTest
             // shared across the whole engine, if multiple tests are run in parallel, the saving could cause a crash
             // in the unit tests.
             AZ::UserSettingsComponentRequestBus::Broadcast(&AZ::UserSettingsComponentRequests::DisableSaveOnFinalize);
+
+            // Create and register a test device attribute for evaluating rules
+            AZStd::string name { "TestAttribute" };
+            AZStd::string description { "Description" };
+            AZStd::string valueString{ "ValidAttribute" };
+            AZStd::any value{ valueString };
+            TestDeviceAttribute::EvalFunc eval = [valueString](AZStd::string_view rule)
+            {
+                auto regex = AZStd::regex(rule.data(), rule.size());
+                return AZStd::regex_match(valueString, regex);
+            };
+            m_testDeviceAttribute = AZStd::make_shared<TestDeviceAttribute>(name, description, value, eval);
```

Comment:
nitpick: You can move the name, description, value and eval function into the TestDeviceAttribute constructor.
```suggestion
            m_testDeviceAttribute = AZStd::make_shared<TestDeviceAttribute>(AZStd::move(name), AZStd::move(description), AZStd::move(value), AZStd::move(eval));
```